lib/evaluate.js

```diff
@@ -239,6 +240,19 @@ export const runFraudDetection = async (roundIndex, measurements, sparkRoundDeta
     }
   }
 
+  //
+  // 3. Reward only first TN measurements
+  //
+  const tasksPerNode = new Map()
+  for (const m of measurements) {
+    if (m.fraudAssessment && m.fraudAssessment !== 'OK') continue
+    const node = `${m.inet_group}::${m.participantAddress}`
+    tasksPerNode.set(node, (tasksPerNode.get(node) ?? 0) + 1)
+    if (tasksPerNode.get(node) > sparkRoundDetails.maxTasksPerNode) {
```

Comment:
BTW, historic rounds cached at the CDN layer will not have `maxTasksPerNode` (it will be `undefined`). This will happen when we do a dry run using the latest spark-evaluate code against an older round.

I think we don't need to worry about that; just be aware.